jbmc/src/java_bytecode/simple_method_stubbing.cpp

```diff
@@ -170,7 +170,8 @@ void java_simple_method_stubst::create_method_stub(symbolt &symbol)
       symbol_table);
     const symbol_exprt &init_symbol_expression = init_symbol.symbol_expr();
     code_assignt get_argument(
-      init_symbol_expression, symbol_exprt(this_argument.get_identifier()));
+      init_symbol_expression,
+      symbol_exprt(this_argument.get_identifier(), this_argument.type()));
     get_argument.add_source_location() = synthesized_source_location;
```

Comment:
This one does appear to be a bugfix!

Reply:
Yes, but it should actually get merged via #3747, I'll then rebase this PR on top of it.